internal/controllers/clusterresourceset/clusterresourceset_scope.go

```diff
@@ -79,11 +80,11 @@ func newResourceReconcileScope(
 
 	switch addonsv1.ClusterResourceSetStrategy(clusterResourceSet.Spec.Strategy) {
 	case addonsv1.ClusterResourceSetStrategyApplyOnce:
-		return &reconcileApplyOnceScope{base}
+		return &reconcileApplyOnceScope{base}, nil
 	case addonsv1.ClusterResourceSetStrategyReconcile:
-		return &reconcileStrategyScope{base}
+		return &reconcileStrategyScope{base}, nil
 	default:
-		return nil
+		return nil, fmt.Errorf("unsupported or empty resource strategy: %q", clusterResourceSet.Spec.Strategy)
```

Comment:
```suggestion
		return nil, errors.Errorf("unsupported or empty resource strategy: %q", clusterResourceSet.Spec.Strategy)
```
(CAPI still uses "github.com/pkg/errors")